### ISSUE TITLE
About model Training

### ISSUE THREAD

**Issue by WhiteHe001** [zhangtaolab/Plant_DNA_LLMs#7]
I am a newcomer to deep learning, and when I used your provided example to fine-tune my plant gene data, I found that it overfitted and had a not high accuracy. Is there any good solution for this? Thank you for your help

====================================================================
 root@ab5e9b91d0c6:~/Plant_DNA_LLMs-main# python model_finetune.py     --model_name_or_path ../plant-dnabert-BPE       --train_data  train_data.csv      --test_data   test_data.csv     --eval_data   val_data.csv     --train_task classification     --labels 'Not promoter;Core promoter'     --run_name plant_dnagpt_BPE_promoter     --per_device_train_batch_size 4     --per_device_eval_batch_size 8     --learning_rate 1e-5     --num_train_epochs 5     --load_best_model_at_end     --metric_for_best_model 'f1'     --save_strategy epoch     --logging_strategy epoch     --evaluation_strategy epoch     --output_dir plant-dnagpt-BPE-promoter
=================================================================
Mamba-ssm is not installed on the machine, cannot run DNA Mamba momdel.
/root/miniconda/lib/python3.8/site-packages/transformers/training_args.py:1568: FutureWarning: `evaluation_strategy` is deprecated and will be removed in version 4.46 of  Transformers. Use `eval_strategy` instead
  warnings.warn(
Map: 100%|██████████████████████████████████████████████████████████████████████████████████████████████| 42904/42904 [00:24<00:00, 1774.08 examples/s]
Map: 100%|████████████████████████████████████████████████████████████████████████████████████████████████| 5363/5363 [00:03<00:00, 1772.56 examples/s]
Map: 100%|████████████████████████████████████████████████████████████████████████████████████████████████| 5363/5363 [00:02<00:00, 1875.17 examples/s]
Some weights of BertForSequenceClassification were not initialized from the model checkpoint at ../plant-dnabert-BPE and are newly initialized: ['bert.pooler.dense.bias', 'bert.pooler.dense.weight', 'classifier.bias', 'classifier.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
model_finetune.py:603: FutureWarning: `tokenizer` is deprecated and will be removed in version 5.0.0 for `Trainer.__init__`. Use `processing_class` instead.
  trainer = Trainer(model=model, tokenizer=tokenizer,
{'loss': 0.5352, 'grad_norm': 1.3934035301208496, 'learning_rate': 8.007465472191117e-06, 'epoch': 1.0}                                                
{'eval_loss': 0.5056552290916443, 'eval_accuracy': 0.8636956927093045, 'eval_f1': 0.8708708708708708, 'eval_precision': 0.8350271002710027, 'eval_recall': 0.9099298634182356, 'eval_matthews_correlation': 0.7301194148029406, 'eval_runtime': 26.4801, 'eval_samples_per_second': 202.53, 'eval_steps_per_second': 25.34, 'epoch': 1.0}

{'loss': 0.4666, 'grad_norm': 0.16128842532634735, 'learning_rate': 6.005599104143337e-06, 'epoch': 2.0}                                               
{'eval_loss': 0.5413329601287842, 'eval_accuracy': 0.8739511467462241, 'eval_f1': 0.879156238827315, 'eval_precision': 0.8523396880415944, 'eval_recall': 0.9077150239940938, 'eval_matthews_correlation': 0.7493242204985134, 'eval_runtime': 26.5139, 'eval_samples_per_second': 202.271, 'eval_steps_per_second': 25.307, 'epoch': 2.0}

{'loss': 0.4079, 'grad_norm': 646.7416381835938, 'learning_rate': 4.0037327360955585e-06, 'epoch': 3.0}                                                
{'eval_loss': 0.5829598903656006, 'eval_accuracy': 0.880850270371061, 'eval_f1': 0.883839301945101, 'eval_precision': 0.870702005730659, 'eval_recall': 0.8973791066814323, 'eval_matthews_correlation': 0.7619651976556434, 'eval_runtime': 26.5003, 'eval_samples_per_second': 202.375, 'eval_steps_per_second': 25.32, 'epoch': 3.0}

{'loss': 0.3656, 'grad_norm': 0.13039427995681763, 'learning_rate': 2.0018663680477792e-06, 'epoch': 4.0}                                              
{'eval_loss': 0.6324098110198975, 'eval_accuracy': 0.8789856423643483, 'eval_f1': 0.8843343432543219, 'eval_precision': 0.8549276361130255, 'eval_recall': 0.9158361018826136, 'eval_matthews_correlation': 0.7597407561248722, 'eval_runtime': 26.4813, 'eval_samples_per_second': 202.52, 'eval_steps_per_second': 25.339, 'epoch': 4.0}

{'loss': 0.3275, 'grad_norm': 0.11055156588554382, 'learning_rate': 0.0, 'epoch': 5.0}                                                                 
{'eval_loss': 0.6835732460021973, 'eval_accuracy': 0.8791721051650195, 'eval_f1': 0.8832432432432433, 'eval_precision': 0.8627243928194298, 'eval_recall': 0.9047619047619048, 'eval_matthews_correlation': 0.7591181438981199, 'eval_runtime': 26.4939, 'eval_samples_per_second': 202.424, 'eval_steps_per_second': 25.327, 'epoch': 5.0}

{'train_runtime': 3944.0646, 'train_samples_per_second': 54.391, 'train_steps_per_second': 13.598, 'train_loss': 0.42056269065093, 'epoch': 5.0}

**Comment by ljq12697** [comment]
You can try to change the default hyperparameters such as '--learning_rate', 'weight_decay', 'warmup_steps', etc., to make the performance better of the fine-tuned model and avoid overfitting.
Detailed description of these hyperparameters can be referred here: https://huggingface.co/docs/transformers/en/main_classes/trainer#transformers.TrainingArguments

**Comment by ljq12697** [comment]
I have seen your training results, the accuracy, F1 score and matthews_correlation show the performance of the fine-tuned model. I think this performance is not bad.
You may see that the train loss is decreasing while the eval loss is increasing, which means that the model may have overfitting.
The learning rate, warmup ratio/step, hyperparameter of the Adam optimizer all influence the training results, perhaps you can try to change these hyperparameters and re-train on your data to avoid overfitting.
